codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -447,6 +449,46 @@ impl SystemConfig {
         })
     }
 
+    fn load_env_vars(project_root: &PathBuf) -> Result<EnvMap> {
+        let env_map = match EnvLoader::with_path(project_root.join(".env"))
+            .sequence(EnvSequence::InputThenEnv)
+            .load()
+        {
+            Ok(env_map) => env_map,
+            // It'll fail if .env is missing, so we return a map with std::env only
+            Err(dotenvy::Error::Io(_, _)) => EnvLoader::new()
+                .sequence(EnvSequence::EnvOnly)
+                .load()
+                .context("Failed loading environment variables")?,
+            err => err.context("Failed loading .env file")?,
+        };
+        Ok(env_map)
+    }
+
+    fn interpolation_config_variables(config_string: String, env_map: EnvMap) -> Result<String> {
+        let mut missing_vars = Vec::new();
+
+        let re = Regex::new(r"\$\{([a-zA-Z_][0-9a-zA-Z_]*)\}").unwrap();
+        let config_string = re.replace_all(&config_string, |caps: &Captures| {
+            match env_map.var(&caps[1]) {
+                Ok(val) => val,
+                Err(_) => {
+                    missing_vars.push(caps[1].to_string());
+                    (&caps[0]).to_string()
```

Comment:
What is the difference here between using cas[1] and caps[0]?

Reply:
caps[0] is the original content, caps[1] is the content of the first regex group (what's inside of the paranthes)